api/base/pagination.py

```diff
@@ -128,6 +128,11 @@ def paginate_queryset(self, queryset, request, view=None):
             return super(JSONAPIPagination, self).paginate_queryset(queryset, request, view=None)
 
 
+class MaxSizePagination(JSONAPIPagination):
+    page_size = 1000
```

Comment:
Can we not have just made the page_size get set to the length of the queryset results (or equivalent)? Thus not much of a max but a full results set on a single page...